bayesian/utils.py

```diff
@@ -64,15 +63,18 @@ def get_user_email(user_profile):
         return default_email
 
 
-def server_create_component_bookkeeping(ecosystem, name, version, user_profile):
+def create_component_bookkeeping(ecosystem, packages_list,
+                                 user_id, user_agent, manifest_hash, request_id):
     """Run the component analysis for given ecosystem+package+version."""
     args = {
-        'external_request_id': uuid.uuid4().hex,
+        'external_request_id': request_id,
         'data': {
-            'api_name': 'component_analyses',
-            'user_email': get_user_email(user_profile),
-            'user_profile': user_profile,
-            'request': {'ecosystem': ecosystem, 'name': name, 'version': version}
+            'api_name': 'component_analyses_post',
+            'manifest_hash': hashlib.md5(manifest_hash.encode()).hexdigest(),
```

Comment:
The field name in the request says manifest_hash, then why are computing again here?

Reply:
Manifeat_hash will actually contain the file path.

Reply:
when in the header.

Reply:
it should be fixed.

also it needs api spec change